### PR TITLE
Move linter names to Linter::Collection

### PR DESCRIPTION
**Pull request by jcmorrow**
Move linter names to Linter::Collection and rename `BETA_LANGUAGES` to `BETA_LINTERS`. This is a change isolated from https://github.com/houndci/hound/pull/1149

### CODE REVIEW

File: app/models/linter/collection.rb [hunk 1]
```diff
@@ -14,6 +14,10 @@ class Collection
       Linter::Swift,
     ].freeze
 
+    def self.linter_names
+      LINTERS.map { |klass| klass.name.demodulize.downcase }
```
Comment (gylaz) [inline app/models/linter/collection.rb]: It's `self.class.name.demodulize.underscore` in `app/models/linter/base.rb`. I'm not sure which one we *should* use, but it seems they should be consistent.

File: app/models/hound_config.rb [hunk 2]
```diff
@@ -1,22 +1,12 @@
...
-    go
-    haml
-    javascript
-    python
-    ruby
-    scss
-    swift
-  )
+  ).freeze
+  CONFIG_FILE = ".hound.yml".freeze
```
Comment (gylaz) [inline app/models/hound_config.rb]: How about using `# frozen_string_literal: true` a the top of the file?

**Comment by gylaz** [comment]
Two questions, then LGTM.

File: app/models/linter/collection.rb (hunk 1 above)
Comment (jcmorrow) [inline app/models/linter/collection.rb]: Yeah `underscore` is superior here.

File: app/models/hound_config.rb [hunk 3]
```diff
@@ -1,22 +1,13 @@
+# frozen_string_literal: true
```
Comment (jcmorrow) [inline app/models/hound_config.rb]: @gylaz does it go on the top line outside of the class or inside of the class?

Comment (gylaz) [inline app/models/hound_config.rb]: This is correct. First line in the file

**Pull request merged**